vck/src/commonMain/kotlin/at/asitplus/wallet/lib/data/rfc/tokenStatusList/StatusList.kt

```diff
@@ -0,0 +1,61 @@
+package at.asitplus.wallet.lib.data.rfc.tokenStatusList
+
+import at.asitplus.wallet.lib.DefaultZlibService
+import at.asitplus.wallet.lib.ZlibService
+import at.asitplus.wallet.lib.data.rfc.tokenStatusList.primitives.TokenStatusBitSize
+import kotlinx.serialization.Serializable
+
+/**
+ * specification: https://www.ietf.org/archive/id/draft-ietf-oauth-status-list-06.html#name-status-list
+ * Status list in its compressed form.
+ */
+@Serializable(with = StatusListSerializer::class)
+data class StatusList(
+    val compressed: ByteArray,
+    val statusBitSize: TokenStatusBitSize,
+    val aggregationUri: String?,
+) {
+    constructor(
+        view: StatusListView,
+        aggregationUri: String?,
+        zlibService: ZlibService = DefaultZlibService(),
+    ) : this(
+        compressed = zlibService.compress(view.uncompressed)
+            ?: throw IllegalStateException("Member `uncompressed` must be compressible."),
+        statusBitSize = view.statusBitSize,
+        aggregationUri = aggregationUri,
+    )
+
+    private var view: StatusListView? = null
```

Comment:
Can we refactor this mutable state away ... by using a lazy property or something else?
(If it makes this task easier, we can always use DefaultZlibService, because it's implemented on all platforms in Kotlin code anyway)

Reply:
fixed in 93ce860696aa4077acb9a053a8b3eac9bee2733b